hack/istio/create-mesh-namespace.sh

```diff
@@ -0,0 +1,164 @@
+#/bin/bash
+
+# This creates/deletes a mesh-enabled namespace/project
+
+# Given a namepace, prepare it for inclusion in Maistra's control plane
+# This means:
+# 1. Create a SMM
+prepare_maistra() {
+  local ns="${1}"
+
+  cat <<EOM | ${CLIENT_EXE} apply -f -
+apiVersion: maistra.io/v1
+kind: ServiceMeshMember
+metadata:
+  name: default
+  namespace: ${ns}
+spec:
+  controlPlaneRef:
+    namespace: ${ISTIO_NAMESPACE}
+    name: "$(${CLIENT_EXE} get smcp -n ${ISTIO_NAMESPACE} -o jsonpath='{.items[0].metadata.name}' )"
+EOM
+}
+
+: ${CLIENT_EXE:=oc}
+: ${DELETE_NAMESPACE:=false}
+: ${ENABLE_INJECTION:=true}
+: ${ISTIO_NAMESPACE:=istio-system}
+: ${SERVICE_ACCOUNTS:="default"}
+
+while [ $# -gt 0 ]; do
+  key="$1"
+  case $key in
+    -c|--client)
+      CLIENT_EXE="$2"
+      shift;shift
+      ;;
+    -d|--delete)
+      DELETE_NAMESPACE="$2"
+      shift;shift
+      ;;
+    -ei|--enable-injection)
+      ENABLE_INJECTION="$2"
+      shift;shift
+      ;;
+    -in|--istio-namespace)
+      ISTIO_NAMESPACE="$2"
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -c|--client: either 'oc' or 'kubectl'
+  -d|--delete: either 'true' or 'false'. If 'true' the namespace will be deleted, not installed.
+  -ei|--enable-injection: either 'true' or 'false' (default is true). If 'true' auto-inject proxies for the workloads.
+  -in|--istio-namespace <name>: Where the Istio control plane is installed (default: istio-system).
+  -n|--namespace <name>: The namespace to be created or deleted. REQUIRED.
+  -sa|--service-accounts
+      The service accounts that will be able to authenticate with docker with the credentials you provide.
+      You can specify more than one service account separated with spaces.
+      If you are using this to be able to pull Istio addon images, this value should be "prometheus grafana default".
```

Comment:
at least pretend you didn't plagiarize :)

Reply:
imitation is flattery!